src/string_immutable.zig

```diff
@@ -1227,12 +1227,38 @@ pub fn utf16Codepoint(comptime Type: type, input: Type) UTF16Replacement {
     }
 }
 
+pub fn convertUTF16ToUTF8(list_: std.ArrayList(u8), comptime Type: type, utf16: Type) !std.ArrayList(u8) {
+    var list = list_;
+    const length = bun.simdutf.length.utf8.from.utf16.le(utf16);
+    try list.ensureTotalCapacityPrecise(length);
+
+    var remaining_input = utf16;
+    var start: usize = 0;
+
+    const replacement_char = [_]u8{ 239, 191, 189 };
+    var result = bun.simdutf.convert.utf16.to.utf8.with_errors.le(remaining_input, list.items.ptr[start..list.capacity]);
+    list.items.len = result.count;
+    while (result.status == .surrogate) {
+        try list.resize(list.items.len + 3);
```

Comment:
This reallocates the array to that specific size

Instead, we want `ensureUnusedCapacity(3)` and then to set `items.len += 3`